src/beams/interpolation.hpp

```diff
@@ -166,7 +166,8 @@ inline std::vector<double> GenerateGLLPoints(const size_t order) {
     }
 
     const size_t n_nodes = order + 1;
-    std::vector<double> gll_points(n_nodes, 0.);
+    auto gll_points = std::vector<double>(n_nodes, 0.);
+    gll_points.resize(n_nodes);
```

Comment:
Is this resizing necessary here, since we already initialize the `std::vector`?

Reply:
It shouldn't be.  In fact, it's a no-op.  I was getting compiler warnings locally (gcc 12.3) from the `.front()` and `.back()` calls saying that it was dereferencing a null pointer.  Obviously, the warnings were incorrect, but calling resize here silences them.  I wish I knew a better way.